arch/cpu/nrf52840/Makefile.nrf52840

```diff
@@ -0,0 +1,144 @@
+LDSCRIPT ?= $(CONTIKI_CPU)/ld/nrf52840-bootloader.ld
+CFLAGS += -DBOARD_PCA10059
+else
+LDSCRIPT ?= $(CONTIKI_CPU)/ld/nrf52840.ld
+CFLAGS += -DBOARD_PCA10056
+endif
+
+### CPU-dependent directories
+CONTIKI_CPU_DIRS += . dev rf
+
+### CPU-dependent source files
+CONTIKI_CPU_SOURCEFILES += clock.c rtimer-arch.c uart0.c dbg.c watchdog.c
+CONTIKI_CPU_SOURCEFILES += int-master.c slip-arch.c
+CONTIKI_CPU_SOURCEFILES += nrf52840-ieee.c
+
+CONTIKI_SOURCEFILES += $(CONTIKI_CPU_SOURCEFILES)
+
+CFLAGS += -DNRF52840_XXAA
+CFLAGS += -D__HEAP_SIZE=512
+CFLAGS += -DSWI_DISABLE0
+CFLAGS += -DCONFIG_GPIO_AS_PINRESET
+CFLAGS += -ggdb
+CFLAGS += -mfloat-abi=hard -mfpu=fpv4-sp-d16
+
+LDFLAGS += -mabi=aapcs -mfloat-abi=hard -mfpu=fpv4-sp-d16
+LDFLAGS += --specs=nano.specs -lc -lnosys
+LDFLAGS += -L $(NRF5_SDK_ROOT)/modules/nrfx/mdk
+
+# nRF52 SDK sources
+NRF52_SDK_C_SRCS += components/boards/boards.c \
+  components/libraries/util/app_error.c \
+  components/libraries/util/app_error_handler_gcc.c \
+  components/libraries/util/app_error_weak.c \
+  components/libraries/util/nrf_assert.c \
+  components/libraries/util/app_util_platform.c \
+  components/libraries/queue/nrf_queue.c \
+  components/libraries/strerror/nrf_strerror.c \
+  modules/nrfx/drivers/src/nrfx_clock.c \
+  modules/nrfx/drivers/src/nrfx_rtc.c \
+  modules/nrfx/drivers/src/nrfx_gpiote.c \
+  modules/nrfx/drivers/src/nrfx_rng.c \
+  modules/nrfx/drivers/src/nrfx_wdt.c \
+  modules/nrfx/mdk/system_nrf52840.c \
+  integration/nrfx/legacy/nrf_drv_clock.c \
+  integration/nrfx/legacy/nrf_drv_rng.c \
+  components/libraries/fifo/app_fifo.c \
+
+NRF52_SDK_ASM_SRCS = modules/nrfx/mdk/gcc_startup_nrf52840.S
+
+CONTIKI_SOURCEFILES += $(notdir $(NRF52_SDK_C_SRCS))
+CONTIKI_SOURCEFILES += $(notdir $(NRF52_SDK_ASM_SRCS))
+
+# Include folders common to all targets
+NRF52_SDK_INC_PATHS += modules/nrfx
+NRF52_SDK_INC_PATHS += modules/nrfx/mdk
+NRF52_SDK_INC_PATHS += modules/nrfx/hal
+NRF52_SDK_INC_PATHS += modules/nrfx/soc
+NRF52_SDK_INC_PATHS += modules/nrfx/drivers/src
+NRF52_SDK_INC_PATHS += modules/nrfx/drivers/include
+NRF52_SDK_INC_PATHS += components/libraries/experimental_section_vars
+NRF52_SDK_INC_PATHS += components/libraries/strerror
+NRF52_SDK_INC_PATHS += components/libraries/delay
+NRF52_SDK_INC_PATHS += components/libraries/util
+NRF52_SDK_INC_PATHS += components/boards
+NRF52_SDK_INC_PATHS += integration/nrfx
+NRF52_SDK_INC_PATHS += integration/nrfx/legacy
+NRF52_SDK_INC_PATHS += components/libraries/queue
+NRF52_SDK_INC_PATHS += components/libraries/fifo
+NRF52_SDK_INC_PATHS += components/libraries/log
+NRF52_SDK_INC_PATHS += components/libraries/log/src
+NRF52_SDK_INC_PATHS += components/drivers_nrf/nrf_soc_nosd
+
+EXTERNALDIRS += $(addprefix $(NRF5_SDK_ROOT)/, $(NRF52_SDK_INC_PATHS))
+
+include $(ARCH_PATH)/cpu/arm/cortex-m/cm4/Makefile.cm4
+
+# Upload via J-Link using nrfjprog 
+
+ifdef NRF52_UPLOAD_SN
+	NRFJPROG_FLAGS += --snr $(NRF52_UPLOAD_SN)
+endif
+
+%.upload: $(OUT_HEX) %.$(TARGET)
```

Comment:
I wonder if some of those virtual targets used for programming devices can be made `.PHONY`

Reply:
Added `nrf-upload-sequence` as `.PHONY`